test/helpers/generators.go

```diff
@@ -121,3 +141,74 @@ func GenerateGatewayConfiguration(gatewayConfigurationNSN types.NamespacedName)
 		},
 	}
 }
+
+// GenerateHTTPRoute generates an HTTPRoute to be used in tests
+func GenerateHTTPRoute(namespace string, gatewayName, serviceName string, opts ...func(*gatewayv1.HTTPRoute)) *gatewayv1.HTTPRoute {
+	httpRoute := &gatewayv1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      uuid.NewString(),
+			Annotations: map[string]string{
+				"konghq.com/strip-path": "true",
+			},
+		},
+		Spec: gatewayv1.HTTPRouteSpec{
+			CommonRouteSpec: gatewayv1.CommonRouteSpec{
+				ParentRefs: []gatewayv1.ParentReference{{
+					Name: gatewayv1.ObjectName(gatewayName),
+				}},
+			},
+			Rules: []gatewayv1.HTTPRouteRule{
+				{
+					Matches: []gatewayv1.HTTPRouteMatch{
+						{
+							Path: &gatewayv1.HTTPPathMatch{
+								Type:  lo.ToPtr(gatewayv1.PathMatchPathPrefix),
+								Value: lo.ToPtr("/test"),
+							},
+						},
+					},
+					BackendRefs: []gatewayv1.HTTPBackendRef{
+						{
+							BackendRef: gatewayv1.BackendRef{
+								BackendObjectReference: gatewayv1.BackendObjectReference{
+									Name: gatewayv1.ObjectName(serviceName),
+									Port: lo.ToPtr(gatewayv1.PortNumber(80)),
+									Kind: lo.ToPtr(gatewayv1.Kind("Service")),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, opt := range opts {
+		opt(httpRoute)
+	}
+
+	return httpRoute
+}
+
+// MustGenerateTLSSecret generates a TLS secret to be used in tests
+func MustGenerateTLSSecret(t *testing.T, namespace, secretName string, hosts []string) *corev1.Secret {
```

Comment:
Nit: `secretName` is not really necessary and could could just use a `GenerateName` given that the secret is returned anyway right?

Do look at `test/integration/test_httproute.go` line 150 where this is provided the `host` for some reason which seems weird.